content/src/content/jcr_root/apps/core/wcm/components/commons/editor/dialog/childreneditor/v1/childreneditor/clientlibs/js/childreneditor.js

```diff
@@ -185,18 +185,23 @@
 
                                 var insertComponentDialog = $(document).find(selectors.insertComponentDialog.self)[0];
                                 var selectList = insertComponentDialog.querySelectorAll(selectors.insertComponentDialog.selectList)[0];
-
                                 // next frame to ensure we remove the default event handler
```

Comment:
I'd simplify as follows:
```
function getSelectListChangeEvent(onCloud) {
}
...
function getSelectListItems(event, onCloud) {
...
}

var onCloud = selectList.toString() === "Coral.List";
...
selectList.off(getSelectListChangeEvent(onCloud);
selectList.on(getSelectListChangeEvent(onCloud) + NS, function(event) {
...
var components = getSelectListItems(event, onCloud);
...
channel.one("coral-overlay:beforeclose", function() {
    selectList.off(getSelectListChangeEvent(onCloud) + NS);
...
```

Reply:
@jckautzmann  the line :

`selectList.on(getSelectListChangeEvent(onCloud) + NS, function(event) {`

does not take into account the addtional second argument for the onCloud case, so it would have to be set to : 

`selectList.on(getSelectListChangeEvent(onCloud) + NS, onCloud ? "coral-list-item" : null, function(event) {`

Would that be ok ?

Reply:
@ issues addressed please check again.

Reply:
we could introduce a specific method for the 2nd argument, e.g. `getSelectListSelector(onCloud)`.

Reply:
@jckautzmann added `getSelectListSelector` function.